recipes-external/external-toolchain/external-toolchain.bb

```diff
@@ -0,0 +1,30 @@
+SDKPATHTOOLCHAIN ?= "${SDKPATH}/toolchain"
+
+inherit cross-canadian
+
+# FIXME: Set to the union of all included component licenses. Ideally this would
+# adapt to the components utilized. Alternatively, package up each component of
+# the toolchain separately.
+LICENSE = "CLOSED"
+
+PN .= "-${TRANSLATED_TARGET_ARCH}"
+SKIPPED = "1"
+SKIPPED:tcmode-external = "0"
+
+python () {
+    external = d.getVar("EXTERNAL_TOOLCHAIN")
+    if not external or not os.path.isdir(external) or d.getVar("SKIPPED") == "1":
+        raise bb.parse.SkipRecipe("An existing external toolchain at EXTERNAL_TOOLCHAIN is required and TCMODE must be set appropriately")
+}
+
+deltask do_configure
+deltask do_compile
+
+do_install () {
+    bbnote "Copying ${EXTERNAL_TOOLCHAIN}/. to ${D}/${SDKPATHTOOLCHAIN}/"
+    install -d ${D}/$(dirname ${SDKPATHTOOLCHAIN})
+    cp -a "${EXTERNAL_TOOLCHAIN}/." "${D}/${SDKPATHTOOLCHAIN}/"
```

Comment:
we don't need this to be recursive? i.e. -r

Reply:
`cp -a` is the same as `cp -dR --preserve=all`, it's already recursive.